### PR TITLE
2405: Enable CI

### PR DESCRIPTION
**Pull request by makubacki**
## Description

This PR intends to establish the minimal set of changes necessary
to enable (mostly) feature agnostic CI on the release/202405 branch.

The goal is to put the foundation in place, so future changes have
the toolchain and plugin changes to test their code reducing the
need for future rework of their changes.

---

### High-Level Changes

- VS2022 Support
- CI and infrastructure related file addition and changes to support
  build and CI
- Plugin changes for the latest stuart tools and to contain the changes
  necessary to support checking current repo contents

### Future Work

- Enable ARM and AARCH64 architectures in pipelines
  - These need additional work and changes that will be focused on in
    a future PR
- Enable CryptoPkg in pipelines
  - Build issues are ignored for now since Mu uses Shared Crypto which
    will significantly change CryptoPkg content

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [x] Includes documentation?

## How This Was Tested

- Local CI & pipelines

## Integration Instructions

- N/A - Local repo CI

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

**Review by kenlautner — APPROVED** [review]

File: BaseTools/Plugin/WindowsVsToolChain/WindowsVsToolChain.py [hunk 1]
```diff
@@ -173,27 +176,175 @@ def do_pre_build(self, thebuilder):
...
+
         return 0
 
     def _get_vs_install_path(self, vs_version, varname):
         # check if already specified
         path = None
         if varname is not None:
             path = shell_environment.GetEnvironment().get_shell_var(varname)
-
+
```
Comment (os-d) [inline BaseTools/Plugin/WindowsVsToolChain/WindowsVsToolChain.py]: nit: whitespace addition

File: CryptoPkg/CryptoPkg.dsc [hunk 2]
```diff
@@ -115,8 +115,10 @@
   RngLib|MdePkg/Library/BaseRngLib/BaseRngLib.inf
 
 [LibraryClasses.ARM, LibraryClasses.AARCH64]
```
Comment (os-d) [inline CryptoPkg/CryptoPkg.dsc:117]: nit: commit name still references StandaloneMmPkg.

**Review by os-d — APPROVED** [review]

File: BaseTools/Plugin/WindowsVsToolChain/WindowsVsToolChain.py (hunk 1 above)
Comment (makubacki) [reply BaseTools/Plugin/WindowsVsToolChain/WindowsVsToolChain.py]: Removed in latest push.

File: CryptoPkg/CryptoPkg.dsc (hunk 2 above)
Comment (makubacki) [reply CryptoPkg/CryptoPkg.dsc:117]: Updated in latest push.

**Pull request merged**